src/Maestro/API/V1/Blocks.hs

```diff
@@ -0,0 +1,21 @@
+module Maestro.API.V1.Blocks where
+
+import           Maestro.Types.V1
+import           Servant.API
+import           Servant.API.Generic
+
+data BlocksAPI route = BlocksAPI
+  { -- | Get details of the specified block by hash
+    _blockByHash
+      :: route
+      :- "blocks"
+      :> Capture "hash_or_height" BlockHash
```

Comment:
Yes! I thought about this other option too. For some reason, it felt cleaner for me, to have two separate functions instead of adding a new data type.